models/appointment_resource_planning_model.py

```diff
@@ -38,60 +39,60 @@ def get_appointments_for_date(cls, config, date, survey_tla, questionnaires):
+                dh.PrimaryKeyValue AS CaseReference,
+                ExtractValue(dh.AdditionalData, "//Field[@Name='CATIAppointment.WhoFor']/@Value") AS RespondentName,
+                ExtractValue(dh.AdditionalData, "//Field[@Name='CATIAppointment.ClctNum']/@Value") AS TelephoneNumber,
                 TIME_FORMAT(dbci.AppointmentStartTime, "%H:%i") AS AppointmentTime,
-                CASE
-                   WHEN
-                      dbci.GroupName = "TNS"
-                      OR dbci.SelectFields LIKE '%<Field FieldName="QDataBag.IntGroup">TNS</Field>%'
-                      OR dh.AdditionalData LIKE '%<Field Name="QDataBag.IntGroup" Status="Response" Value="\\'TNS\\'"/>%'
-                   THEN
-                      "Other"
-                   WHEN
-                      dbci.GroupName = "WLS"
-                      OR dbci.SelectFields LIKE '%<Field FieldName="QDataBag.IntGroup">WLS</Field>%'
-                      OR dh.AdditionalData LIKE '%<Field Name="QDataBag.IntGroup" Status="Response" Value="\\'WLS\\'"/>%'
-                   THEN
-                      "Welsh"
-                   ELSE
-                      "English"
+            CASE
+                WHEN
+                    dbci.GroupName = "TNS"
+                    OR ExtractValue(dbci.SelectFields, "//QDataBag.IntGroup") = "TNS"
+                    OR ExtractValue(dh.AdditionalData, "//Field[@Name='QDataBag.IntGroup']/@Value") = "TNS"
+                THEN
+                    "Other"
+                WHEN
+                    dbci.GroupName = "WLS"
+                    OR ExtractValue(dbci.SelectFields, "//QDataBag.IntGroup") = "WLS"
+                    OR ExtractValue(dh.AdditionalData, "//Field[@Name='QDataBag.IntGroup']/@Value") = "WLS"
+                THEN
+                    "Welsh"
+                ELSE
+                    "English"
                 END
-                AS AppointmentLanguage,
-                COUNT(*) AS Total    
-            FROM
-                cati.DaybatchCaseInfo AS dbci
-            LEFT JOIN DialHistory dh
-                ON dh.InstrumentId = dbci.InstrumentId
-                AND dh.PrimaryKeyValue = dbci.PrimaryKeyValue
-                AND dh.DialResult = "Appointment"    
-            INNER JOIN UniqueDialHistoryIdTable uid
-                ON dh.id = uid.id
-            WHERE
-               dbci.AppointmentType != "0"
-               AND dbci.AppointmentStartDate LIKE "{date}%"
+            AS AppointmentLanguage
+        
+            FROM cati.DaybatchCaseInfo AS dbci
+                LEFT JOIN DialHistory dh
+                    ON dh.InstrumentId = dbci.InstrumentId
+                    AND dh.PrimaryKeyValue = dbci.PrimaryKeyValue
+                    AND dh.DialResult = "Appointment"
+                INNER JOIN UniqueDialHistoryIdTable uid
+                    ON dh.id = uid.id
+            WHERE dbci.AppointmentType != "0"
+            AND dbci.AppointmentStartDate LIKE "{date}%"
             GROUP BY
```

Comment:
Group by is now redundant as no longer using an aggregate.

Reply:
You're absolutely right, I've made the amendments and successfully tested.